### ISSUE TITLE
Svelte 5 runes mode compatibility

### ISSUE THREAD

**Issue by HummingMind** [lucide-icons/lucide#2312]
### Package

- [ ] lucide
- [ ] lucide-angular
- [ ] lucide-flutter
- [ ] lucide-preact
- [ ] lucide-react
- [ ] lucide-react-native
- [ ] lucide-solid
- [X] lucide-svelte
- [ ] lucide-vue
- [ ] lucide-vue-next
- [ ] Figma plugin
- [ ] all JS packages
- [ ] site

### Description

A .next version of lucide-svelte is needed that adds compatibility with runes mode for Svelte 5 RC (and soon Svelte 5 final).
The current version does not work, as it uses $$props instead of the new $porps() rune.

Thank you!

### Use cases

Needed for Svelte 5 RC support of runes mode

### Checklist

- [X] I have searched the existing issues to make sure this bug has not already been reported.

**Comment by HummingMind** [comment]
@ericfennis understandable. It would need to be a new major version, which is why I suggested a ".next" version. Could be publishes side-by-side with the current version. 🍺 

If it is not time yet, I get it. 👍🏻

**Comment by Shyrogan** [comment]
Moving towards Svelte 5 would also be a great option to implement some features, see #1302 for example.

If you need someone to write the code for it, I would love to do it. Just want to make sure those changes would be merged if someone writes the code for it.

**Comment by alissa-tung** [comment]
some like

```svelte
<script lang="ts">
  import defaultAttributes from './defaultAttributes';
  import type { IconNode } from './types';
  import type { SvelteHTMLElements } from 'svelte/elements';

  const {
    class: className,
    children,

    name = undefined,
    color = 'currentColor',
    size = 24,
    strokeWidth = 2,
    absoluteStrokeWidth = false,
    iconNode = [],

    ...props
  }: SvelteHTMLElements['svg'] & {
    name: string | undefined;
    color: string;
    size: number | string;
    strokeWidth: number | string;
    absoluteStrokeWidth: boolean;
    iconNode: IconNode;
  } = $props();

  const mergeClasses = <ClassType = string | undefined | null,>(...classes: ClassType[]) =>
    classes
      .filter((className, index, array) => {
        return Boolean(className) && array.indexOf(className) === index;
      })
      .join(' ');
</script>

<svg
  {...defaultAttributes}
  {...props}
  width={size}
  height={size}
  stroke={color}
  stroke-width={absoluteStrokeWidth ? (Number(strokeWidth) * 24) / Number(size) : strokeWidth}
  class={mergeClasses('lucide-icon', 'lucide', name ? `lucide-${name}` : '', className)}
>
  {#each iconNode as [tag, attrs]}
    <svelte:element
      this={tag}
      {...attrs}
    />
  {/each}
  {@render children?.()}
</svg>

```

should work

**Comment by alissa-tung** [comment]
https://github.com/lucide-icons/lucide/pull/2380/files

**Comment by Shyrogan** [comment]
> https://github.com/lucide-icons/lucide/pull/2380/files

Should fix what I mentioned. Thank you @alissa-tung, I don't see any improvements to make, let's hope it gets merged :)

**Comment by alissa-tung** [comment]
> > https://github.com/lucide-icons/lucide/pull/2380/files
> 
> Should fix what I mentioned. Thank you @alissa-tung, I don't see any improvements to make, let's hope it gets merged :)

I think that would be great, but since I am new to JavaScript, I think I can not even handle my own pull request. I opened it hope someone can notice that there should happen something for Svelte 5 version, and if someone can take over this PR I thing would be much helpful

**Comment by bobvork** [comment]
Please note that `svelte:component` has been deprecated as of `5.0.0-next.217`: https://github.com/sveltejs/svelte/releases/tag/svelte%405.0.0-next.217. It should work for now, but might be worth looking into the new syntax, as found in the preview docs here: https://svelte-5-preview.vercel.app/docs/breaking-changes#breaking-changes-in-runes-mode-svelte-component-is-no-longer-necessary

**Comment by kirso** [comment]
Also type compatibility:
```ts
import type { Component } from 'svelte';
import { Globe, Map, CheckCircle, FileText, Clock, Users, Search } from 'lucide-svelte';
import type { Icon } from 'lucide-svelte';
export const Icons: Component<Icon> = {
	Globe,
	Map,
	CheckCircle,
	FileText,
	Clock,
	Users,
	Search
};

export { Globe, Map, CheckCircle, FileText, Clock, Users, Search };
```

Throws: 
Object literal may only specify known properties, and 'Globe' does not exist in type 'Component<Icon, {}, string>'.ts(2353)

**Comment by cowboycodr** [comment]
https://github.com/lucide-icons/lucide/issues/2312#issuecomment-2361496558
Typing seems to be the biggest issue currently.

**Comment by KaiStarkk** [comment]
Most people searching for this are probably trying to create a dynamic Icon component to speed up development. If so, this is the current way to do it I believe.

```svelte
<!-- lib/components/Icon.svelte -->
<script lang="ts">
	import * as icons from 'lucide-svelte';
	import type { Component } from 'svelte';

	type IconName = keyof typeof icons;

	let { name, ...others }: { name: IconName } = $props();

	let Icon = icons[name] as unknown as Component;
</script>

<Icon {...others} />
```

```svelte
<!-- Usage in SomeComponent.svelte -->
<script lang="ts">
	import Icon from '$lib/components/Icon.svelte';
</script>

<Icon name="Disk" />
```

Just not sure if this enables vite to perform tree-shaking.

**Comment by sug48** [comment]
I just upgraded to Svelte 5 and it seems like the compiler is doing something funky to Lucide when I'm running runes mode. I keep getting the error `Cannot use `$$props` in runes mode [plugin vite-plugin-svelte:optimize-svelte]`

So for example, this is what one of my `.svelte` files looks like: 
```
<script>
import BoomBox from 'lucide-svelte/icons/boom-box'
<script>

<BoomBox size="16" color="var(--color-text-medium-dark)"></BoomBox>
```

And then the icon is getting compiled as:
```
<Icon name="boom-box" {...$$props} iconNode={iconNode}>
  <slot/>
</Icon>
```

**Comment by HummingMind** [comment]
Lucide is not compatible with Steve 5 runes-only mode yet. That was the whole point of this issue. You either need to remove the runes only compiler option or wait until new major version of Lucide.

**Comment by erwinschaap** [comment]
Also waiting for the update! Would be nice if Lucide will be compatible with Svelte 5 runes-only mode

**Comment by sug48** [comment]
> Lucide is not compatible with Steve 5 runes-only mode yet. That was the whole point of this issue. You either need to remove the runes only compiler option or wait until new major version of Lucide.

Fair enough and I figured as much. But I also figured it would be worthwhile to document an error for others encountering the same thing.

**Comment by matheustavarestrindade** [comment]
Hello guys! if you are having problems with the new version of svelte 5, just update the typing like so:

`<script lang="ts">
	import type { Icon as LucideIcon } from 'lucide-svelte';

	type Props = {
		Icon: typeof LucideIcon;
	};

	const { Icon }: Props = $props();
</script>

<Icon />
`
Then you can just pass the icon component as usual,

`Icon={User} `  or something :)

**Comment by ithan** [comment]
Hello guys! For anyone that does not want to make a separate component for this, you can also use snippets like:

```
<script lang="ts">
  import type { Icon as LucideIcon } from 'lucide-svelte';
</script>

{#snippet render_icon(icon: typeof LucideIcon)}
  {@const Icon = icon} <!-- This is here for better syntax highlighting in vscode, but you don't technically need it -->
  <Icon class="h-3.5 w-3.5" />
{/snippet}

{@render render_icon(icon)}
```

**Comment by niemyjski** [comment]
Any idea when this will be supported? If someone needs backwards compatibility with the old version, then they can stay on an older version.

**Comment by shenshouer** [comment]
https://github.com/lucide-icons/lucide/issues/2850

**Comment by keshmirian** [comment]
Nice one, @ithan. Exactly what I've been searching for!

**Comment by ericfennis** [comment]
Fixed in #2753
